Update contributor docs to remove old GCC information (#31667)
yugabyte/yugabyte-db: docs/content/stable/contribute/core-database/build-and-test.md

Specify the compiler using the `--gcc<version_number>` and `--clang<version_number>` flags.
The specific versions supported can be found in `./build-support/third-party-archives.yml`, which details the configurations regularly tested in-house.

> Pre-built toolchains are downloaded to `/opt/yb-build` and used by default. It is possible to instead build with a GCC toolchain installed on the system by setting the `YB_USE_SYSTEM_GCC` environmental variable:

Reviewer: ```suggestion
Pre-built toolchains are downloaded to `/opt/yb-build` and used by default. To instead build with a GCC toolchain installed on the system, set the `YB_USE_SYSTEM_GCC` environment variable:

```
